src/main.rs

```diff
@@ -71,7 +71,11 @@ fn setup(config_folder: &Path) -> IoResult<(Data<RwLock<Config>>, ServerConfig,
     // Create necessary directories.
     fs::create_dir_all(&server_config.upload_path)?;
     for paste_type in &[PasteType::Url, PasteType::Oneshot, PasteType::OneshotUrl] {
-        fs::create_dir_all(paste_type.get_path(&server_config.upload_path))?;
+        fs::create_dir_all(
+            paste_type
+                .get_path(&server_config.upload_path)
+                .expect("failed to get path for paste type"),
+        )?;
```

Comment:
There is still a possible panic here.

Sorry for all my comments, but your code introduces a possible panic, while the "old" code just didn't create the dir. Maybe I am missing panics in the "old" code along the path, e.g. when rustypaste tries to save a file when the dir does not exist, but I think it just returns an error message.

Reply:
There is - yes. I missed this when I wrote the last commit. I'll map it to a sensible error type so it won't panic